### DISCUSSION TITLE
For Heap out of memory due to java.util.concurrent.ConcurrentHashMap, is that just due to build graph being too big?

### DISCUSSION THREAD

**Discussion by Ryang20718** [bazelbuild/bazel#23401] [Q&A] [answered]
For Heap out of memory due to java.util.concurrent.ConcurrentHashMap, is that just due to build graph being too big?
<img width="933" alt="image" src="https://github.com/user-attachments/assets/9e8654d4-6e9d-4207-960a-15176fb04e36">

**Comment by meisterT** [comment, marked as answer]
Yes, that map is essentially the backing data structure of [Skyframe](https://bazel.build/contribute/codebase#skyframe) - https://bazel.build/rules/performance#enabling-memory-tracking might help you to find out surprising things about your build graph composition.
